StellarApp/Controllers/NewsViewController.swift

```diff
@@ -56,20 +60,24 @@ class NewsViewController: UIViewController {
         
     }
     private func setGradient() {
-        let firstColor = UIColor.init(red: 236/255, green: 233/255, blue: 230/255, alpha: 1)
-        let secondColor = UIColor.init(red: 255/255, green: 255/255, blue: 255/255, alpha: 1)
+        let firstColor = #colorLiteral(red: 0.8039215803, green: 0.8039215803, blue: 0.8039215803, alpha: 1)
+        let secondColor = #colorLiteral(red: 0.2549019754, green: 0.2745098174, blue: 0.3019607961, alpha: 1)
         let gradient = CAGradientLayer()
         gradient.frame = self.newsView.newsCollectionView.bounds
         gradient.colors = [firstColor.cgColor,secondColor.cgColor]
         self.newsView.layer.insertSublayer(gradient, at: 0)
         newsView.newsCollectionView.delegate = self
     }
-    private func getArticles(keyword: String) {
-        ApiClient.getNews(query: keyword) { (error, data) in
+    private func getArticles(keyword: String, pageNumber: Int) {
+        ApiClient.getNews(query: keyword, page: pageNumber) { (error, data) in
             if let error = error {
                 print(error.errorMessage())
             } else if let data = data {
+                if self.articles.isEmpty {
                 self.articles = data
+                } else {
```

Comment:
this is where you should batch update the collection view
right now you're reloading the whole thing

Reply:
https://developer.apple.com/documentation/uikit/uicollectionview/1618097-insertitems

Reply:
https://developer.apple.com/documentation/uikit/uicollectionview/1618045-performbatchupdates